packages/vite/src/node/plugins/importAnalysisBuild.ts

```diff
@@ -454,25 +472,56 @@ export function buildImportAnalysisPlugin(config: ResolvedConfig): Plugin {
               }
 
               if (markerStartPos > 0) {
-                s.overwrite(
-                  markerStartPos,
-                  markerStartPos + preloadMarkerWithQuote.length,
-                  // the dep list includes the main chunk, so only need to reload when there are
-                  // actual other deps. Don't include the assets dir if the default asset file names
-                  // are used, the path will be reconstructed by the import preload helper
+                const { modulePreload } = config.build
+                // the dep list includes the main chunk, so only need to reload when there are actual other deps.
+                const depsArray =
                   deps.size > 1 ||
-                    // main chunk is removed
-                    (hasRemovedPureCssChunk && deps.size > 0)
-                    ? `[${[...deps]
-                        .map((d) =>
-                          JSON.stringify(
-                            relativePreloadUrls
-                              ? path.relative(path.dirname(file), d)
-                              : d
+                  // main chunk is removed
+                  (hasRemovedPureCssChunk && deps.size > 0)
+                    ? [...deps]
+                    : []
+                const toRelative = (dep: string) =>
+                  path.relative(path.dirname(file), dep)
+                const resolvedDependencies =
```

Comment:
nit: Maybe it's just me, but I find this ternary to be quite long and not that easy to follow. Consider improving its readability/maintainability by refactoring it (e.g., into a function or even if-else statements).

Reply:
The code changed quite a bit after [9f1eaa5 (#9938)](https://github.com/vitejs/vite/pull/9938/commits/9f1eaa5f7539eb7588506419465aa8c7251b2efd), please check if it is more clear now